public/javascripts/conversation.js

```diff
@@ -1,7 +1,12 @@
 "use strict";
 $(function() {
     var thingpediaUrl = document.body.dataset.thingpediaUrl;
-    var url = 'ws' + location.href.substring('http'.length) + '/../api/conversation';
+    var url = new URL('api/conversation', location.href);
```

Comment:
This will append the url to the location.href minus last part.

ie, running it on this page

```js
> new URL('api/conversation', location.href).toString()
"https://github.com/stanford-oval/almond-server/pull/24/api/conversation"
```

Reply:
Which is correct, because the page where this JS is running is http://127.0.0.1:3000/conversation, and the URL we want is http://127.0.0.1:3000/api/conversation

In HA, it would be http://hassio.local:3123/some-ingress-stuff/conversation --> http://hassio.local:3123/some-ingress-stuff/api/conversation

Unless I misunderstand how ingress works in HA?

Reply:
yeah that is correct. Ok, just wanted to make sure it was intended.